File: RaspberryDashboard-Backend/RaspberryDashboard-Backend/Models/Weather/OpenWeather.cs
using Newtonsoft.Json;

namespace RaspberryDashboard_Backend.Models.Weather
{
    public class OpenWeather
    {
        [JsonProperty("coord")]
        public WeatherCoord Coord { get; set; }
        [JsonProperty("weather")]
        public Weather[] Weather { get; set; }
        [JsonProperty("main")]
        public MainWeather Main { get; set; }
        [JsonProperty("visibility")]
        public int Visibility { get; set; }
        [JsonProperty("wind")]
        public Wind Wind { get; set; }
        [JsonProperty("name")]
        public string Place { get; set; }

        public string Error { get; set; }

    }

    public class Wind
    {
        [JsonProperty("speed")]
        public double Speed { get; set; }
        [JsonProperty("deg")]
        public int Deg { get; set; }
    }

    public class MainWeather
    {
        [JsonProperty("temp")]
        public double Temp { get; set; }
        [JsonProperty("feels_like")]
        public double TempFeel { get; set; }
        [JsonProperty("temp_min")]
        public double TempMin { get; set; }
        [JsonProperty("temp_max")]
        public double TempMax { get; set; }
        [JsonProperty("pressure")]
        public int Pressure { get; set; }
        [JsonProperty("humidity")]
        public int Humidity { get; set; }
    }

    public class Weather
    {
        [JsonProperty("id")]
        public int Id { get; set; }
        [JsonProperty("main")]
        public string Main { get; set; }
        [JsonProperty("description")]
        public string Description { get; set; }
        [JsonProperty("icon")] 
        public string Icon { get; set; }
    }

    public class WeatherCoord
    {
        [JsonProperty("lon")]
        public double Lon { get; set; }
        [JsonProperty("lat")]
        public double Lat { get; set; }
    }

}

File: RaspberryDashboard-Frontend/src/app/model/weather/open-weather.ts
class WeatherCoord {
  public Lon: number;
  public Lat: number;
}

class Weather {
  public Id: number;
  public Main: string;
  public Description: string;
  public Icon: string;
  public IconUrl: string;
}

class MainWeather {
  public Temp: number;
  public TempFeel: number;
  public TempMin: number;
  public TempMax: number;
  public Pressure: number;
  public Humidity: number;
}

class Wind {
  public Speed: number;
  public Deg: number;
}

export class OpenWeather {
  public Coord: WeatherCoord;
  public Weather: Weather[] = [];
  public Main: MainWeather;
  public Visibility: number;
  public Wind: Wind;
  public Place: string;

}
